### PR TITLE
feat: argent hybrid session keys

### PR DESCRIPTION
**Pull request by bluecco**

### CODE REVIEW

File: src/hybridSessionBackendService.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+          r: sessionSignature[0].toString(),
+          s: sessionSignature[1].toString(),
+        },
+      },
+    }
+
+    // TODO: type
+    const body: any = {
+      session,
+    }
```
[the remark this answers is not in the document]
Comment (bluecco) [reply src/hybridSessionBackendService.ts:71]: ah no, totally forgot and will update

File: src/hybridSessionBackendService.ts [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(body),
+    })
+    const json = await response.json()
+
+    return json.signature
+  }
+
+  /* public async signOutsideTxAndSession(
```
[the remark this answers is not in the document]
Comment (bluecco) [reply src/hybridSessionBackendService.ts:138]: yes

File: src/hybridSessionDappService.ts [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+    invocationSignerDetails: InvocationsSignerDetails,
+    sessionTypedData: TypedData,
+  ): Promise<ArraySignatureType> {
+    const compiledCalldata = transaction.getExecuteCalldata(
+      calls,
+      invocationSignerDetails.cairoVersion,
+    )
+    let txHash
+    if (
+      Object.values(RPC.ETransactionVersion2).includes(
```
[the remark this answers is not in the document]
Comment (bluecco) [reply src/hybridSessionDappService.ts:96]: was done by blockchain team, will add
@sgc-code ^

File: src/hybridSessionDappService.ts [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+        invocationSignerDetails.version as any,
+      )
+    ) {
+      const det = invocationSignerDetails as V2InvocationsSignerDetails
```
[the remark this answers is not in the document]
Comment (bluecco) [reply src/hybridSessionDappService.ts:100]: yep

File: src/hybridSessionBackendService.ts [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+    if (
+      Object.values(RPC.ETransactionVersion2).includes(
+        transactionsDetail.version as any,
+      )
+    ) {
+      const txDetailsV2 = transactionsDetail as V2InvocationsSignerDetails
+      apiBaseUrl = this.getApiBaseUrl(txDetailsV2.chainId)
+
+      body.transaction = {
+        // Can be either v1 or v3 transaction
```
[the remark this answers is not in the document]
Comment (bluecco) [reply src/hybridSessionBackendService.ts:82]: just kept backend comment on this one, can be removed 👍

File: src/hybridSessionBackendService.ts [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+      },
+    }
+
+    // TODO: type
+    const body: any = {
+      session,
+    }
+
+    if (
+      Object.values(RPC.ETransactionVersion2).includes(
```
[the remark this answers is not in the document]
Comment (bluecco) [reply src/hybridSessionBackendService.ts:74]: was done by blockchain team, will add
@sgc-code ^

[the remark this answers is not in the document]
Comment (sgc-code) [reply src/hybridSessionBackendService.ts:74]: SN.js naming is a bit weird but ETransactionVersion2 includes tx v0, v1 and v2

[the remark this answers is not in the document]
Comment (vladutjs) [reply src/hybridSessionBackendService.ts:74]: oh nice

**Review by vladutjs — APPROVED** [review]

**Pull request merged**